Allow launcher step to specify SDK version

reborg opened the pull request:
Hi,

I just added one more optional parameter to specify the target SDK to run in the simulator. I also added a skeleton step that makes use of the version so you can write something like "Given I launch the app using iOS 4.2".

reborg committed:
Allow the client to be specific about the sdk to use to launch the app in the simulator

reborg committed:
added launch step with SDK version to the skeleton steps

moredip commented:
This looks good. Would I be able to persuade you to add the ability to configure family (i.e. whether to launch in iphone or ipad)?

reborg committed:
Added additional parameter to specify type of simulator to run, some
tests and refactoring

refactored sim_launcher dependncies into module variables

app_path instance variable clashing with example step renamed

simulator ipad-iphone option now defaults to iphone

added another example step that makes use of the version to launch the simulator

@refactor extracted formatter for path location suggestions

reborg commented:
Here you go. Also added a few tests to drive the small change and a bit of refactoring. Just rake test inside the gem folder.

moredip commented:
Yay tests! This is great. Pulling it in.
